zerver/lib/actions.py

```diff
@@ -1403,6 +1403,7 @@ def send_user_email_update_event(user_profile: UserProfile) -> None:
     )
 
 
+@transaction.atomic
 def do_change_user_delivery_email(user_profile: UserProfile, new_email: str) -> None:
     delete_user_profile_caches([user_profile])
```

Comment:
Do we want this inside the `transaction.atomic` block? This hits memcached, so possibly we should think of this like we do of `send_event` calls, which hit tornado? On the other hand - `.save()` calls can often hit memcached too (due to post_save signal hooks), which is harder to avoid inside transactions and we're already doing that - so perhaps we should just give up on this and accept the fact that we will unavoidably hit memcached inside transactions?

Reply:
Since it's safe to delete something from memcached for any reason at all (it's a cache), I think it's not a bug to do so inside a transaction block like this.